.github/workflows/go.yml

```diff
@@ -0,0 +1,34 @@
+name: Build and test Go
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: "*"
```

Comment:
Here you can target only master, this has the benefit of running the CI only once on the code on a scenario like this:
```
master ---------------
                 /
pr1 ------------/
         /
pr2 ----/
```

With your solution you test on `pr2 -> pr1` and `pr1 -> master` and changing `branches:` to `master` you would build only on `pr1 -> master`.

Reply:
This actually intentional. Unit testing `pr2 -> pr1` is also useful, to catch errors early, no?

Reply:
This was not done previously with travis but if this is a feature you would like to have then we can revert the change and put it back like it was before.